tests/integration/test_verify.py

```diff
@@ -16,6 +19,28 @@ def test_verify_valid_sources(source_dir):
     assert exit_code == 0
 
 
+@pytest.mark.parametrize('source_dir, result_file', [
+    ("tests/test_files/yaml_source_dir/valid_yamls_with_multiple_crds",
+     "tests/test_files/yaml_source_dir/valid_yamls_with_multiple_crds/result/bundle.yaml")
+])
+def test_verify_with_output_file(source_dir, result_file):
+    with TemporaryDirectory() as temp_dir:
+        output = os.path.join(temp_dir, "output.yaml")
+        verify_command = f'operator-courier verify {source_dir} --output={output}'
+        process = subprocess.Popen(verify_command,
+                                   shell=True,
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.STDOUT)
+        exit_code = process.wait()
+        assert exit_code == 0
+
+        file_created = os.path.isfile(output)
+        assert file_created
+
+        content_correct = filecmp.cmp(result_file, output)
```

Comment:
This naive comparison will be pretty hard to achieve as there is no order guaranteed in python dicts and depends on python version/implementation

Reply:
You're probably right but that is literally what we do right now for all of our tests. At the moment ordering seems consistent simply because these tests always pass (and because we are just copying blobs). But adding code that reads these files and checks their content is just more code that needs to be tested itself, which I am against -- that's why the original implementation I did doesn't bother to look at the file contents, just that the file itself was created. IMO if we wanted to solve that problem we would have to invest more significant effort in solving the problem in the general case, or just ignore it.

My preference is for now either remove this assertion, or keep it and remove it if we get in a case where the result is not guaranteed (due to some other change in the source).

Reply:
You can try to loaf that out file and expected file via yaml and compare those 2 objects at python level

Reply:
Yes but the minute we do that we need to codify what data is in the files in these tests. So if the logic about how these files are created is changed, then we need to change these tests. That is very brittle and IMO should be avoided.

Reply:
I don't understand what do you mean. If you change something in code you have to update `result/bundle.yaml` anyway. I still don't see where is the issue between
```python
assert load_file(output) == load_file('result/bundle.yaml')
```
and
```python
assert yaml.safe_load(load_file(output)) == yaml.safe_load(load_file('result/bundle.yaml'))
```

Reply:
Ah I see your point now. Yeah that's a great idea, it resolves the problem without worrying about the schema.

Reply:
Code updated, please review when you have time. Thanks Martin.

Reply:
Yeah, this didn't help. Order of processing of CRD files is different on ubuntu vs macos. And I forgot that it is yaml inside yaml, so without deeper deconstruction those data cannot be validated in this way.

Probably just load output file decode with yaml module and check id format is valid yaml and nonempty.

Reply:
For inspiration: https://github.com/release-engineering/operators-manifests-push-service/blob/master/tests/integration/utils.py#L26

Reply:
@MartinBasti But that test will break the minute the format changes. I specifically want to avoid encoding the data format of the bundle into the tests.